### ISSUE TITLE
Support autocomplete

### ISSUE THREAD

**Issue by kpj** [rr-/urwid_readline#1]
Hello,

I am using your library in one of my [projects](https://github.com/kpj/Vydia/blob/e5e7ddadec910df9359af9a578660d073effb1f1/vydia/core/view.py#L51) and would greatly appreciate support for input auto-completion.
As you mention it in your README, is there any possibility of implementing this feature?

Regards,
kpj

**Comment by rr-** [comment]
Can you outline API and keypresses you'd be interested in?

**Comment by kpj** [comment]
I thought of a behavior similar to [readline](https://docs.python.org/3/library/readline.html), i.e. (in the simplest case) autocomplete on `<TAB>` using something like:
```python
commands = ['start', 'stop', 'next']
def compl(text, state):
  tmp = [c for c in commands if c and c.startswith(text)] if text else commands
  try:
    return tmp[state]
  except IndexError:
    return None

w = urwid_readline.ReadlineEdit()
w.enable_autocomplete(compl)
```

which could internally include:

```python
import readline
[..]
class ReadlineEdit(urwid.Edit):
[..]
def enable_autocomplete(self, func):
  def display_matches(self, substitution, matches, longest_match_length):
    [..]
  readline.set_completer(func)
  readline.parse_and_bind('tab: complete')
  readline.set_completer_delims(' \t\n;')
  readline.set_completion_display_matches_hook(display_matches)
[..]
```

`display_matches` would need to be adapted to `ReadlineEdit`. (Maybe using various calls to `set_edit_text`?)

**Comment by rr-** [comment]
A proof of concept is available on https://github.com/rr-/urwid_readline/tree/autocomplete branch. Please let me know if it satisfies your criteria.

**Comment by kpj** [comment]
Very cool, I like it a lot!
It seems to work as expected. I am now using it [here](https://github.com/kpj/Vydia/blob/5c01535ec144c539b640f2075d2d87eae417e1f7/vydia/core/view.py#L72).

One recommendation: add `self._autocomplete_func = None` to `ReadlineEdit`'s `__init__` function.
It otherwise crashes when not calling `enable_autocomplete`.

**Comment by rr-** [comment]
Done, released as 0.5, if there are any more problems LMK and I'll fix them.

Cool project, thanks for pointing me to it :)
